src/python_src/util/app_utilities.py

```diff
@@ -153,44 +224,152 @@ def download_ml_models_from_s3(model_file: str, vectorizer_file: str) -> tuple[s
+need_download = False
+
+if import_time_download_enabled and (not os.path.exists(model_file) or not os.path.exists(vectorizer_file)):
+    need_download = True
+    logging.info("Missing model files - will download from S3")
+elif import_time_download_enabled and sha_check_enabled:
+    # Verify existing files if SHA checking is enabled
+    chunk_size = app_config["ml_classifier"]["integrity_verification"]["hash_config"]["chunk_size_bytes"]
+    expected_model_sha = app_config["ml_classifier"]["integrity_verification"]["expected_checksums"]["model"]
+    expected_vectorizer_sha = app_config["ml_classifier"]["integrity_verification"]["expected_checksums"]["vectorizer"]
+
+    # Allow environment variables to override config values
+    expected_model_sha = os.environ.get("ML_MODEL_SHA256", expected_model_sha)
+    expected_vectorizer_sha = os.environ.get("ML_VECTORIZER_SHA256", expected_vectorizer_sha)
+
+    logging.info("Verifying SHA-256 of existing model files")
+    logging.info(f"Expected model SHA-256: {expected_model_sha}")
+    logging.info(f"Expected vectorizer SHA-256: {expected_vectorizer_sha}")
+
+    model_valid = verify_file_sha256(model_file, expected_model_sha, chunk_size)
+    vectorizer_valid = verify_file_sha256(vectorizer_file, expected_vectorizer_sha, chunk_size)
+
+    if not model_valid or not vectorizer_valid:
+        logging.warning("Existing model files failed SHA-256 verification - will re-download from S3")
+        # Remove invalid files
+        if not model_valid and os.path.exists(model_file):
+            os.remove(model_file)
+        if not vectorizer_valid and os.path.exists(vectorizer_file):
+            os.remove(vectorizer_file)
+        need_download = True
+    else:
+        logging.info("All existing model files passed SHA-256 verification - skipping download")
+elif import_time_download_enabled:
+    # Files exist but SHA checking is disabled
+    logging.info("Model files exist and SHA-256 verification is disabled - skipping download")
+
+# download all files from S3 if needed
+if need_download:
     os.makedirs(model_directory, exist_ok=True)
-    download_ml_models_from_s3(model_file, vectorizer_file)
+    try:
+        download_ml_models_from_s3(model_file, vectorizer_file)
+        logging.info("Successfully downloaded ML models from S3")
+    except Exception as e:
+        logging.warning(f"Failed to download ML models from S3: {e}")
+        # Check if files exist locally as fallback
+        if os.path.exists(model_file) and os.path.exists(vectorizer_file):
+            logging.info("Using existing local ML model files as fallback")
+        else:
+            logging.warning("ML classifier will not be available - no local files found")
 
 ml_classifier = None
-if os.path.exists(model_file) and os.path.exists(vectorizer_file):
-    ml_classifier = MLClassifier(model_file, vectorizer_file)
+if import_time_download_enabled and os.path.exists(model_file) and os.path.exists(vectorizer_file):
```

Comment:
```suggestion
if os.path.exists(model_file) and os.path.exists(vectorizer_file):
```
question: should we remove the `import_time_download_enabled` at this point?  would it be sufficient to check the existence of the 2 files?  why: i wonder about the use case where i have the models downloaded and have set `import_time_download_enabled = false`; are we saying the app should refuse to instantiate the ML classifier under those circumstances?

Reply:
This was in-case there were moments where the AWS access may not be working, or setups where that may not need the ML-Classifier.